include/message_filters/cache.h

```diff
@@ -272,12 +272,12 @@ class Cache : public SimpleFilter<M>
 
     MConstPtr out ;
 
-    int i=cache_.size()-1 ;
+    size_t i=cache_.size()-1 ;
```

Comment:
This change makes the condition in the while loop (`i>=0`) obsolete. When `cache_` is empty a rollover would occur.

Reply:
Yes, I'll go back and verify anywhere that I made signedness changes.